worlds/messenger/transitions.py

```diff
@@ -0,0 +1,96 @@
+from typing import TYPE_CHECKING
+
+from BaseClasses import Region
+from entrance_rando import EntranceType, randomize_entrances
+from .connections import RANDOMIZED_CONNECTIONS, TRANSITIONS
+from .options import ShuffleTransitions, TransitionPlando
+
+if TYPE_CHECKING:
+    from . import MessengerWorld
+
+
+def connect_plando(world: "MessengerWorld", plando_connections: TransitionPlando) -> None:
+    def remove_dangling_exit(region: Region) -> None:
+        # find the disconnected exit and remove references to it
+        for _exit in region.exits:
+            if not _exit.connected_region:
+                break
+        else:
+            raise ValueError(f"Unable to find randomized transition for {plando_connection}")
+        region.exits.remove(_exit)
+
+    def remove_dangling_entrance(region: Region) -> None:
+        # find the disconnected entrance and remove references to it
+        for _entrance in region.entrances:
+            if not _entrance.parent_region:
+                break
+        else:
+            raise ValueError(f"Invalid target region for {plando_connection}")
+        region.entrances.remove(_entrance)
+
+    multiworld = world.multiworld
+    player = world.player
+    for plando_connection in plando_connections:
+        # get the connecting regions
+        reg1 = multiworld.get_region(plando_connection.entrance, player)
+        reg2 = multiworld.get_region(plando_connection.exit, player)
```

Comment:
Up to you if you want to use these, I just saw you changed an instance in another file. There's also three others later in this file
```suggestion
        reg1 = world.get_region(plando_connection.entrance)
        reg2 = world.get_region(plando_connection.exit)
```